code/game/machinery/computer/communications.dm

```diff
@@ -130,6 +131,32 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 				message_admins("[key_name_admin(usr)] has sent a message to the other server.")
 				CM.lastTimeUsed = world.time
 
+		if("purchase_menu")
+			state = STATE_PURCHASE
+
+		if("buyshuttle")
+			if(authenticated==2)
+				var/list/shuttles = flatten_list(shuttle_templates)
+				var/datum/map_template/shuttle/S = locate(href_list["chosen_shuttle"]) in shuttles
+				if(S && istype(S))
+					for(var/obj/machinery/shuttle_manipulator/M in machines)

```

Comment:
Huh?

There's only one shuttle manipulator isn't there, just store a reference to it in SSshuttle

come on kor this is bad and you know better than that.